src/components/sectionList/SectionListWrapper.tsx

```diff
@@ -95,11 +97,13 @@ export const SectionListWrapper = ({
         (model: BaseListModel) => (
             <DefaultListActions
                 model={model}
+                modelType={schema.displayName}
                 onShowDetailsClick={handleDetailsClick}
                 onOpenSharingClick={setSharingDialogId}
+                onDeleteSuccess={refetch}
```

Comment:
We were talking about context in the meeting today. I actually think putting the `refetch` in context could make sense here. This is imo a good use-case for it. It's not going to change, and it's proper dependency injection.
The outer list (eg. DataElement/List, or generic-list) would render the Provider with the refetch. Then we could do 
Could also be in `SectionListWrapper`

Reply:
I'm not sure about this. I was thinking about context for the model, because that's domain complexity we can't do anything about and mixing in the concept of model in deeper components seems unavoidable. So I don't mind complecting those components because it's happening regardless of how we do it.

With `refetch` I have two concerns:

1. It's only being passed down to a single component so far (`<DeleteAction />`). so adding several layers of complexity seems a bit overkill at this point
2. This is the more important point: Using `refetch` explicitly in the `DeleteAction` component would make that component aware of the data-concept which it doesn't really care about conceptually. That's why I added the `onDeleteSuccess` prop intentionally. Everything data-fetching related can stay in the `SectionListWrapper`.

I think I see where you're coming from, it'd be a lot easier to not have to pass down props like `onDeleteSuccess` two or three times. From my experience the resulting complexity isn't worth it because it makes reasoning about the code harder and changing the code more complex (not necessarily easie, but more trivial).

I certainly wouldn't mind revisiting this once it becomes a nuisance to pass down the `refetch` function to too many places and that in itself introduces structural complexity. Maybe it'd make sense to have a `AfterModelManipulationContext` that doesn't explicitly mention the concept of `refetch` (or `data`-fetching in general). Right now I think it's too early to introduce new layers of indirection though.